packages/components/src/combobox-control/test/index.js

```diff
@@ -0,0 +1,141 @@
+	{ name: 'New Zealand Standard Time', abbreviation: 'NST' },
+	{ name: 'Midway Islands Time', abbreviation: 'MIT' },
+	{ name: 'Hawaii Standard Time', abbreviation: 'HST' },
+	{ name: 'Alaska Standard Time', abbreviation: 'AST' },
+	{ name: 'Pacific Standard Time', abbreviation: 'PST' },
+	{ name: 'Phoenix Standard Time', abbreviation: 'PNT' },
+	{ name: 'Mountain Standard Time', abbreviation: 'MST' },
+	{ name: 'Central Standard Time', abbreviation: 'CST' },
+	{ name: 'Eastern Standard Time', abbreviation: 'EST' },
+	{ name: 'Indiana Eastern Standard Time', abbreviation: 'IET' },
+	{ name: 'Puerto Rico and US Virgin Islands Time', abbreviation: 'PRT' },
+	{ name: 'Canada Newfoundland Time', abbreviation: 'CNT' },
+	{ name: 'Argentina Standard Time', abbreviation: 'AGT' },
+	{ name: 'Brazil Eastern Time', abbreviation: 'BET' },
+	{ name: 'Central African Time', abbreviation: 'CAT' },
+];
+const mapTimezoneOption = ( tz ) => ( {
+	value: tz.abbreviation,
+	label: tz.name,
+} );
+
+const timezoneOptions = timezones.map( mapTimezoneOption );
+const defaultLabelText = 'Select a timezone';
+const getLabel = () => screen.getByText( defaultLabelText );
+const getInput = () => screen.getByRole( 'combobox' );
+const getRenderedOptions = () => screen.getAllByRole( 'option' );
+
+describe( 'ComboboxControl', () => {
+	// Use real timers to avoid timeout errors from userEvent
+	beforeEach( () => {
+		jest.useRealTimers();
+	} );
+
+	const TestComboboxControl = ( props ) => {
+		const [ value, setValue ] = useState( null );
+		return (
+			<>
+				<ComboboxControl
+					{ ...props }
+					value={ value }
+					label={ defaultLabelText }
+					options={ timezoneOptions }
+					onChange={ setValue }
+				/>
+				<p data-testid="value-output">{ value }</p>
+			</>
+		);
+	};
+
+	it( 'should render', () => {
+		render( <TestComboboxControl /> );
+
+		const input = getInput();
+		expect( input ).toBeVisible();
+	} );
+
+	it( 'should render with visible label', () => {
+		render( <TestComboboxControl /> );
+		const label = getLabel();
+		expect( label ).toBeVisible();
+	} );
+
+	it( 'should render with hidden label', () => {
+		render( <TestComboboxControl hideLabelFromVision={ true } /> );
+		const label = getLabel();
+
+		expect( label ).toBeInTheDocument();
+		expect( label ).toHaveAttribute(
+			'data-wp-component',
+			'VisuallyHidden'
+		);
+	} );
+
+	it( 'should render with the correct options', () => {
+		render( <TestComboboxControl /> );
+
+		getInput().focus();
+
+		const renderedOptions = getRenderedOptions();
+
+		expect( renderedOptions.length ).toEqual( timezones.length );
+
+		// Confirm the rendered options match the provided dataset.
+		const renderedOptionNames = [];
+		for ( const option of renderedOptions.values() ) {
+			renderedOptionNames.push( option.textContent );
+		}
+		const timezoneNames = timezones.map( ( tz ) => tz.name );
+
+		expect( renderedOptionNames ).toEqual( timezoneNames );
```

Comment:
If I understand what you're trying to achieve here, it could be written in a simpler way:

```suggestion
		// Confirm the rendered options match the provided dataset.
		expect( renderedOptions ).toHaveLength( timezones.length );
		renderedOptions.forEach( ( option, optionIndex) => {
			expect( option ).toHaveTextContent( timezones[optionIndex].name )
		} );
```

Reply:
Much cleaner, thank you!